docs/index.rst

```diff
@@ -0,0 +1,99 @@
+.. vSphere Plugin documentation master file, created by
+   sphinx-quickstart on Thu Dec  8 15:54:24 2016.
+   You can adapt this file completely to your liking, but it should at least
+   contain the root `toctree` directive.
+
+Cloudify vSphere Plugin documentation
+=====================================
+The vSphere plugin allows users to use a vSphere based infrastructure for deploying services and applications.
+
+Contents:
+
+.. toctree::
+    :maxdepth: 3
+
+    types
+    examples
+
+
+Plugin Requirements
+-------------------
+
+Python
+~~~~~~
+* 2.7.x
+
+vSphere Environment
+~~~~~~~~~~~~~~~~~~~
+You will require a working vSphere environment. The plugin was tested with version 5.5.
+
+SSH Keys
+~~~~~~~~
+
+.. highlight:: bash
+
+* You will need SSH keys generated for both the manager and the application VM's. If you are using the default key locations in the inputs, these can be created with the following commands::
+
+    ssh-keygen -b2048 -N "" -q -f ~/.ssh/cloudify-manager-kp.pem
+    ssh-keygen -b2048 -N "" -q -f ~/.ssh/cloudify-agent-kp.pem
+
+* A working vSphere environment. The plugin was tested with version 5.5.
+
+Permissions on vCenter
+~~~~~~~~~~~~~~~~~~~~~~
+
+* To create and destroy virtual machines and storage:
+
+  * On the datacenter (Must Propagate to children):
+
+    * Datastore/Allocate Space
+    * Network/Assign Network
+    * Virtual Machine/Configuration/Add new disk
+    * Virtual Machine/Configuration/Add or remove device
+    * Virtual Machine/Configuration/Change CPU count
+    * Virtual Machine/Configuration/Memory
+    * Virtual Machine/Configuration/Remove disk
+    * Virtual Machine/Interaction/Power On
+    * Virtual Machine/Interaction/Power Off
+    * Virtual Machine/Inventory/Create from existing
+    * Virtual Machine/Inventory/Remove
+
+  * On the specific resource pool:
+
+    * Full permissions recommended
+
+  * On the template(s) to be used
+
+    * Virtual Machine/Provisioning/Customize
+    * Virtual Machine/Provisioning/Deploy template
+
+* To create and destroy port groups:
+
+  * On the datacenter:
+
+    * Host/Configuration/Network configuration
+* To create and destroy distributed port groups:
+
+  * On the datacenter:
+
+    * dvPort group/Create
+    * dvPort group/Delete
+
+
+OS Templates
+~~~~~~~~~~~~
+
+* You need two OS templates of your preferred operating systems (e.g. Ubuntu Trusty) within the vSphere datastores. One for the Cloudify manager and one for the application VMs. The application VM template should accept the Cloudify agent public key for its root user. The Cloudify manager template must accept the cloudify manager public key. Note that you can choose to use same template for both the manager and the application VMs, in that case the shared template must accept both public keys.
```

Comment:
Can we change the e.g. to Centos 7? (Ubuntu won't work for the manager any more)